### DISCUSSION TITLE
Window not responding when server is under heavy load

### DISCUSSION THREAD

**Discussion by ALjungfjar** [tauri-apps/tauri#7652] [Q&A]
Hello, 

I have a Tauri application which is using http REST api for communication between server and client.
I am using an xml library called [xmltree](https://docs.rs/xmltree/latest/xmltree/).
Whenever I send a request telling the server to open the passed xml file path and parse it to an element in memory the window stops responding, as following:
![image](https://github.com/tauri-apps/tauri/assets/93316881/6bd6ced7-c0a5-46e9-90d0-b9db48749b3f)

After the xml element has been parsed the window starts to respond again, so it is obviously the parsing which creates this behavior. 
The request I am doing is asynchronous so client should still be responsive while the server processes the data. To me this looks like the container of the webview stops responding because of the server runtime chewing on some heavy stuff. Is this possible? What should I do to fix this problem? Make the xml parsing use its own thread?

The request I am making from the client is very simple, looks as following:
`fetch(`host/document/${filePath}/open`, {
   method: "POST",
   mode: "no-cors",
})`

**Comment by SpikeHD** [comment]
If the main thread is getting held up by a heavy task (such as your XML parsing), it'll cause holdup on everything else on the main thread (WebView included), so your idea of moving the request and processing to a separate thread is probably the best option.
